From eb1011775f4aa7042bd1b5c671018de76575df7c Mon Sep 17 00:00:00 2001
From: Adam Ford <aford173@gmail.com>
Date: Mon, 30 Nov 2020 09:08:39 -0600
Subject: [PATCH 72/78] arm64: dts: Beacon SOM's:  Increase USDHC1 to support
 up to 200MHz

The WiFi chip is connected to USDHC1 and the pinmux for 100MHz and
200MHz is defined, but they're never referenced.

Adding the references increases throughput.

Before:
[ ID] Interval           Transfer     Bitrate         Retr  Cwnd
[  5]   0.00-2.00   sec  22.5 MBytes  94.3 Mbits/sec   10    165 KBytes
[  5]   2.00-4.00   sec  30.0 MBytes   126 Mbits/sec    6    185 KBytes
[  5]   4.00-6.00   sec  22.4 MBytes  93.8 Mbits/sec    6    173 KBytes
[  5]   6.00-8.00   sec  22.4 MBytes  93.8 Mbits/sec    5    150 KBytes
[  5]   8.00-10.00  sec  22.4 MBytes  93.8 Mbits/sec    3    194 KBytes
[  5]  10.00-12.00  sec  22.4 MBytes  93.8 Mbits/sec    7    175 KBytes
[  5]  12.00-14.00  sec  22.4 MBytes  94.1 Mbits/sec    5    161 KBytes
[  5]  14.00-16.00  sec  22.4 MBytes  93.8 Mbits/sec    7    143 KBytes
[  5]  16.00-18.00  sec  22.4 MBytes  93.8 Mbits/sec    5    185 KBytes
[  5]  18.00-20.00  sec  22.5 MBytes  94.4 Mbits/sec    5    171 KBytes
[  5]  20.00-22.00  sec  22.4 MBytes  93.8 Mbits/sec   11    156 KBytes
[  5]  22.00-24.00  sec  22.4 MBytes  93.8 Mbits/sec    5    136 KBytes
[  5]  24.00-26.00  sec  22.4 MBytes  93.8 Mbits/sec    6    182 KBytes
[  5]  26.00-28.00  sec  22.4 MBytes  93.8 Mbits/sec    7    164 KBytes
[  5]  28.00-30.00  sec  22.4 MBytes  93.8 Mbits/sec   10    147 KBytes
- - - - - - - - - - - - - - - - - - - - - - - - -
[ ID] Interval           Transfer     Bitrate         Retr
[  5]   0.00-30.00  sec   343 MBytes  96.0 Mbits/sec   98             sender
[  5]   0.00-30.00  sec   343 MBytes  95.9 Mbits/sec                  receiver

After:
[ ID] Interval           Transfer     Bitrate         Retr  Cwnd
[  5]   0.00-1.00   sec  20.3 MBytes   171 Mbits/sec    0    830 KBytes
[  5]   1.00-2.00   sec  20.0 MBytes   168 Mbits/sec    0   1.22 MBytes
[  5]   2.00-3.00   sec  22.5 MBytes   189 Mbits/sec    0   1.52 MBytes
[  5]   3.00-4.00   sec  18.8 MBytes   157 Mbits/sec    0   1.99 MBytes
[  5]   4.00-5.00   sec  13.8 MBytes   115 Mbits/sec    0   2.36 MBytes
[  5]   5.00-6.00   sec  13.8 MBytes   115 Mbits/sec    0   2.36 MBytes
[  5]   6.00-7.00   sec  13.8 MBytes   115 Mbits/sec    0   2.53 MBytes
[  5]   7.00-8.00   sec  20.0 MBytes   168 Mbits/sec    0   2.80 MBytes
[  5]   8.00-9.00   sec  20.0 MBytes   168 Mbits/sec    0   3.00 MBytes
[  5]   9.00-10.00  sec  22.5 MBytes   189 Mbits/sec    1   2.10 MBytes
- - - - - - - - - - - - - - - - - - - - - - - - -
[ ID] Interval           Transfer     Bitrate         Retr
[  5]   0.00-10.00  sec   185 MBytes   155 Mbits/sec    1             sender
[  5]   0.00-10.02  sec   184 MBytes   154 Mbits/sec                  receiver

Signed-off-by: Adam Ford <aford173@gmail.com>
---
 arch/arm64/boot/dts/freescale/beacon-imx8mm-som.dtsi | 4 +++-
 arch/arm64/boot/dts/freescale/beacon-imx8mn-som.dtsi | 4 +++-
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/beacon-imx8mm-som.dtsi b/arch/arm64/boot/dts/freescale/beacon-imx8mm-som.dtsi
index a39706c2db35..0512d40c8395 100644
--- a/arch/arm64/boot/dts/freescale/beacon-imx8mm-som.dtsi
+++ b/arch/arm64/boot/dts/freescale/beacon-imx8mm-som.dtsi
@@ -247,8 +247,10 @@
 &usdhc1 {
 	#address-cells = <1>;
 	#size-cells = <0>;
-	pinctrl-names = "default";
+	pinctrl-names = "default", "state_100mhz", "state_200mhz";
 	pinctrl-0 = <&pinctrl_usdhc1>;
+	pinctrl-1 = <&pinctrl_usdhc1_100mhz>;
+	pinctrl-2 = <&pinctrl_usdhc1_200mhz>;
 	bus-width = <4>;
 	non-removable;
 	cap-power-off-card;
diff --git a/arch/arm64/boot/dts/freescale/beacon-imx8mn-som.dtsi b/arch/arm64/boot/dts/freescale/beacon-imx8mn-som.dtsi
index d17b58039440..5ad50c1adcaa 100644
--- a/arch/arm64/boot/dts/freescale/beacon-imx8mn-som.dtsi
+++ b/arch/arm64/boot/dts/freescale/beacon-imx8mn-som.dtsi
@@ -238,8 +238,10 @@
 &usdhc1 {
 	#address-cells = <1>;
 	#size-cells = <0>;
-	pinctrl-names = "default";
+	pinctrl-names = "default", "state_100mhz", "state_200mhz";
 	pinctrl-0 = <&pinctrl_usdhc1>;
+	pinctrl-1 = <&pinctrl_usdhc1_100mhz>;
+	pinctrl-2 = <&pinctrl_usdhc1_200mhz>;
 	bus-width = <4>;
 	non-removable;
 	cap-power-off-card;
-- 
2.17.1

